src/components/AccountDrawer/IconButton.tsx

```diff
@@ -136,41 +142,60 @@ export const IconWithConfirmTextButton = ({
 
   const dimensionsRef = useRef({
     frame: 0,
-    hidden: 0,
+    innerText: 0,
   })
   const dimensions = (() => {
     // once opened, we avoid updating it to prevent constant resize loop
     if (!showText) {
-      dimensionsRef.current = { frame: frameObserver.width || 0, hidden: hiddenObserver.width || 0 }
+      dimensionsRef.current = { frame: frameObserver.width || 0, innerText: hiddenObserver.width || 0 }
     }
     return dimensionsRef.current
   })()
 
   // keyboard action to cancel
   useEffect(() => {
-    if (!showText) return
-    const isClient = typeof window !== 'undefined'
-    if (!isClient) return
-    if (!showText) return
+    if (typeof window === 'undefined') return
+    if (!showText || !frame) return
+
+    const closeAndPrevent = (e: Event) => {
+      setShowText(false)
+      e.preventDefault()
+      e.stopPropagation()
+    }
+
+    const clickHandler = (e: MouseEvent) => {
+      const { target } = e
+      const shouldClose = !(target instanceof HTMLElement) || !frame.contains(target)
+      if (shouldClose) {
+        closeAndPrevent(e)
+      }
+    }
+
     const keyHandler = (e: KeyboardEvent) => {
       if (e.key === 'Escape') {
-        setShowText(false)
-        e.preventDefault()
-        e.stopPropagation()
+        closeAndPrevent(e)
       }
     }
+
+    window.addEventListener('click', clickHandler, { capture: true })
```

Comment:
just checking, did you try the existing `useOnClickOutside` hook and find that it didn't work for this case?

Reply:
oh didn't know we had that